### PR TITLE
chore(kafka): Add ability to read and write headers

### PR DESCRIPTION
**Pull request by Oleksiivanov**
## Description

Added ability to read and write headers

Screenshot 2023-10-17 at 10 40 32

closes #
https://github.com/camunda/connectors/issues/1016

### CODE REVIEW

File: connectors/kafka/src/main/java/io/camunda/connector/kafka/inbound/KafkaInboundMessage.java [hunk 1]
```diff
@@ -40,6 +42,14 @@ public void setValue(Object value) {
...
   }
 
+  public Object getHeaders() {
+    return headers;
+  }
+
+  public void setHeaders(final Object headers) {
+    this.headers = headers;
+  }
+
```
Comment (sbuettner) [inline connectors/kafka/src/main/java/io/camunda/connector/kafka/inbound/KafkaInboundMessage.java:52]: Shouldnt it rather be a Map<String,Object>?

File: connectors/kafka/src/main/java/io/camunda/connector/kafka/inbound/KafkaPropertyTransformer.java [hunk 2]
```diff
@@ -87,6 +83,17 @@ public static KafkaInboundMessage convertConsumerRecordToKafkaInboundMessage(
       ConsumerRecord<String, Object> consumerRecord, ObjectReader objectReader) {
     KafkaInboundMessage kafkaInboundMessage = new KafkaInboundMessage();
     kafkaInboundMessage.setKey(consumerRecord.key());
+
+    if (consumerRecord.headers() != null) {
+      var headerMap = new HashMap<>();
+      for (Header header : consumerRecord.headers()) {
+        headerMap.put(header.key(), new String(header.value(), StandardCharsets.UTF_8));
```
Comment (sbuettner) [inline connectors/kafka/src/main/java/io/camunda/connector/kafka/inbound/KafkaPropertyTransformer.java:90]: Please add to the docs that only UTF-8 strings are supported as header values.

File: connectors/kafka/src/main/java/io/camunda/connector/kafka/outbound/KafkaConnectorFunction.java [hunk 3]
```diff
@@ -79,12 +88,25 @@ private KafkaConnectorResponse executeConnector(final KafkaConnectorRequest requ
...
+              String.valueOf(request.getMessage().getKey()),
+              transformedValue);
+
+      if (request.getHeaders() != null) {
+        for (Map.Entry<String, String> header : request.getHeaders().entrySet()) {
+          producerRecord
+              .headers()
+              .add(header.getKey(), header.getValue().getBytes(StandardCharsets.UTF_8));
+        }
+      }
```
Comment (sbuettner) [inline connectors/kafka/src/main/java/io/camunda/connector/kafka/outbound/KafkaConnectorFunction.java:106]: Please extract into separate method

**Review by sbuettner — APPROVED** [review]
Minor commments

**Review by igpetrov — APPROVED** [review]

**Pull request merged**